asb-native/src/main/java/org/ballerinax/asb/sender/MessageSender.java

```diff
@@ -73,54 +81,71 @@ public MessageSender(String connectionString, String entityType, String topicOrQ
         log.debug("ServiceBusSenderClient initialized");
     }
 
-    /**
-     * Send Message with configurable parameters when Sender Connection is given as
-     * a parameter and
-     * message content as a byte array.
-     *
-     * @param body             Input message content as byte array
-     * @param contentType      Input message content type
-     * @param messageId        Input Message ID
-     * @param to               Input Message to
-     * @param replyTo          Input Message reply to
-     * @param replyToSessionId Identifier of the session to reply to
-     * @param label            Input Message label
-     * @param sessionId        Input Message session Id
-     * @param correlationId    Input Message correlationId
-     * @param timeToLive       Input Message time to live in minutes
-     * @param properties       Input Message properties
-     * @return An error if failed send the message.
-     */
-    public Object send(Object body, Object contentType, Object messageId,
-            Object to, Object replyTo, Object replyToSessionId, Object label, Object sessionId,
-            Object correlationId, Object partitionKey, Object timeToLive, Object properties) {
+    public Object send(BMap<BString, Object> message) {
         try {
-            byte[] byteArray = ((BArray) body).getBytes();
-            ServiceBusMessage asbMessage = new ServiceBusMessage(byteArray);
-            asbMessage.setContentType(ASBUtils.convertString(contentType));
-            asbMessage.setMessageId(ASBUtils.convertString(messageId));
-            asbMessage.setTo(ASBUtils.convertString(to));
-            asbMessage.setReplyTo(ASBUtils.convertString(replyTo));
-            asbMessage.setReplyToSessionId(ASBUtils.convertString(replyToSessionId));
-            asbMessage.setSubject(ASBUtils.convertString(label));
-            asbMessage.setSessionId(ASBUtils.convertString(sessionId));
-            asbMessage.setCorrelationId(ASBUtils.convertString(correlationId));
-            asbMessage.setPartitionKey(ASBUtils.convertString(partitionKey));
-            if (timeToLive != null) {
-                asbMessage.setTimeToLive(Duration.ofSeconds((long) timeToLive));
-            }
-            Map<String, Object> map = ASBUtils.toMap((BMap) properties);
-            asbMessage.getApplicationProperties().putAll(map);
-            sender.sendMessage(asbMessage);
+            ServiceBusMessage messageToSend = constructMessage(message);
+            sender.sendMessage(messageToSend);
             if (log.isDebugEnabled()) {
-                log.debug("Sent the message successfully");
+                log.debug("Sent the message successfully. Message Id = " + messageToSend.getMessageId());
```

Comment:
Diito. String templates can be used.

Reply:
This is Java file. Above comment applies.